### PR TITLE
Sg3 utils bump to 1.42

### PR DESCRIPTION
**Pull request by grueni**

### CODE REVIEW

**Review by AndWac — CHANGES_REQUESTED** [review]
1. Move to appropriate category (eg. sysutils)
2. Make use of $(WS_MAKE_RULES) in Makefile
3. Add REQUIRED_PACKAGES to Makefile

**Review by AndWac — CHANGES_REQUESTED** [review]
This package needs also a new home. As it's also a system package I propose to move it into components/sysutils.
And then squash the commits when finished.

File: components/sg3_utils/Makefile [hunk 1]
```diff
@@ -25,24 +25,30 @@
...
 
-include ../../make-rules/prep.mk
-include ../../make-rules/configure.mk
-include ../../make-rules/ips.mk
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/configure.mk
+include $(WS_MAKE_RULES)/ips.mk
 
 CPPFLAGS += -I$(COMPONENT_DIR)/$(COMPONENT_SRC)/include
+CFLAGS += -D__EXTENSIONS__ -D_XOPEN_SOURCE=1 -D_XOPEN_SOURCE_EXTENDED=1
```
Comment (alarcher) [inline components/sg3_utils/Makefile:47]: Just to make sure that you really want compliance with XPG4v2, right?

File: components/sg3_utils/Makefile [hunk 2]
```diff
@@ -68,4 +74,6 @@ test:           $(TEST_32)
 
 BUILD_PKG_DEPENDENCIES =        $(BUILD_TOOLS)
 
-include ../../make-rules/depend.mk
+include $(WS_MAKE_RULES)/depend.mk
```
Comment (alarcher) [inline components/sg3_utils/Makefile]: You can remove lines 75 and 77.

File: components/sg3_utils/Makefile [hunk 3]
```diff
@@ -25,24 +25,30 @@
...
+COMPONENT_SRC= $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE= $(COMPONENT_SRC).tgz
 COMPONENT_ARCHIVE_HASH= \
-    sha256:3034a4e798404cc963fc46437b0ceeb0edc635e02471ab13aa18acd8b716a27b
-COMPONENT_ARCHIVE_URL=	http://sg.danny.cz/sg/p/$(COMPONENT_ARCHIVE)
-COMPONENT_BUGDB=	utility/sg3utils
+    sha256:1dcb7a0309bd0ba3d4a83acb526973b80106ee26cd9f7398186cd3f0633c9ef3
+COMPONENT_ARCHIVE_URL= http://sg.danny.cz/sg/p/$(COMPONENT_ARCHIVE)
+COMPONENT_LICENSE= GPLv2 
+COMPONENT_LICENSE_FILE= $(COMPONENT_NAME).license
```
Comment (alarcher) [inline components/sg3_utils/Makefile]: You can remove line 39, it is enforced by default.

File: components/sg3_utils/sg3_utils.p5m [hunk 4]
```diff
@@ -24,21 +24,28 @@
 #
 
 <transform file path=usr.*/man/.+ -> default mangler.man.stability uncommitted>
```
Comment (alarcher) [inline components/sg3_utils/sg3_utils.p5m]: I think this transform is applied by default and can be removed.

File: components/sg3_utils/sg3_utils.p5m [hunk 5]
```diff
@@ -137,3 +175,5 @@ license sg3_utils.license license="GPLv2+ and BSD"
 # dependency is here to facilitate the safe transfer of the package 
 # to the 'userland' consolidation.
 depend type=optional fmri=consolidation/desktop/desktop-incorporation@5.12-5.12.0.0.0.4.0
```
Comment (alarcher) [inline components/sg3_utils/sg3_utils.p5m]: Not sure that this dependency still makes sense.

**Review by xen0l — APPROVED** [review]

**Comment by xen0l** [comment]
I'm fine with the changes and will be merging shortly. Thanks @grueni for the change, @AndWac and @alarcher for reviewing.

File: components/sysutils/sg3_utils/Makefile [hunk 6]
```diff
@@ -66,6 +72,4 @@ install:        $(INSTALL_32)
 
 test:           $(TEST_32)
 
-BUILD_PKG_DEPENDENCIES =        $(BUILD_TOOLS)
-
-include ../../make-rules/depend.mk
+EQUIRED_PACKAGES += system/library
```
Comment (pyhalov) [inline components/sysutils/sg3_utils/Makefile:75]: REQUIRED_PACKAGES, not EQUIRED_PACKAGES

**Comment by xen0l** [comment]
Continuted in #2499

**Pull request closed without merging**